CRM/Contact/BAO/Contact.php

```diff
@@ -306,6 +306,11 @@ public static function &create(&$params, $fixAddress = TRUE, $invokeHooks = TRUE
           }
         }
       }
+
+      // CRM-21041: set default 'Communication Style' if unset when creating a contact.
+      if (empty($params['communication_style_id'])) {
```

Comment:
Should also test for the presence of id in params and skip if updating.

Reply:
This whole code block lives inside 
if (empty($params['contact_id'])) { } check
So this will only be executed when creating a contact and will be skipped when updating one.